outbound/adapter/__tests__/pdv-id-outbound-enricher.test.ts

```diff
@@ -0,0 +1,139 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { TelemetryClient } from "applicationinsights";
+import { Client } from "../../../generated/pdv-tokenizer-api/client";
+import { create, MaybePdvDocumentsTypes } from "../pdv-id-outbound-enricher";
+import * as E from "fp-ts/lib/Either";
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { Errors } from "io-ts";
+import { Result } from "../../port/outbound-publisher";
+import {
+  aMockPdvId,
+  aRetrievedProfile,
+  aRetrievedProfileList,
+  aRetrievedServicePreferences,
+  aRetrievedServicePreferencesList
+} from "../../../businesslogic/__mocks__/data.mock";
+
+const mockSave = jest
+  .fn()
+  .mockImplementation(_ =>
+    Promise.resolve(E.right({ status: 200, value: { token: aMockPdvId } }))
+  );
+const mockTrackEvent = jest.fn();
+const mockTelemetryClient = ({
+  trackEvent: mockTrackEvent
+} as unknown) as TelemetryClient;
+const mockTokenizerClient = ({
+  saveUsingPUT: mockSave
+} as unknown) as Client;
+
+const enricher = create(2, mockTokenizerClient, mockTelemetryClient);
+
+describe.each`
+  title                        | value                               | isList   | length
+  ${"profile"}                 | ${aRetrievedProfile}                | ${false} | ${1}
+  ${"profile list"}            | ${aRetrievedProfileList}            | ${true}  | ${2}
+  ${"service preference"}      | ${aRetrievedServicePreferences}     | ${false} | ${1}
+  ${"service preference list"} | ${aRetrievedServicePreferencesList} | ${true}  | ${2}
+`("$title enricher", ({ value, isList, length }) => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const call = isList ? enricher.enrichs(value) : enricher.enrich(value);
+
+  const expectSuccess = (
+    result:
+      | readonly Result<MaybePdvDocumentsTypes>[]
+      | E.Either<Error, MaybePdvDocumentsTypes>
+  ) => {
+    if (isList) {
+      expect(result).toMatchObject([{ success: true }, { success: true }]);
```

Comment:
We can check element in list, too

```suggestion
      expect(result).toMatchObject([
        { success: true, document: { ...value[0], userPDVId: aMockPdvId } },
        { success: true, document: { ...value[1], userPDVId: aMockPdvId } }
      ]);
```

Reply:
check added in https://github.com/pagopa/io-functions-elt/pull/87/commits/0c4d2e0d2acf397d7ec1b02f07837ab6777dbfc1, thank you